src/lib/components/SparklineChart/index.js

```diff
@@ -571,64 +554,19 @@ class SparklineChart extends BaseElement {
     return colorClass;
   }
 
-  /**
-   * (Re)renders the line, gradient. Should be called when .values is changed.
-   * @private
-   */
-  update_() {
-    // Don't do work if no values yet.
-    if (!this.values_.length) {
-      return;
-    }
-
-    const rect = this.getBoundingClientRect();
-    this.width_ = rect.width;
-    this.height_ = rect.height;
-
-    // Account for padding and diameter of data point circle.
-    const circleDiameter = this.circleRadius_ * 2;
-    this.width_ = this.width_ - this.padding_ - circleDiameter;
-    this.height_ =
-      this.height_ - this.padding_ - circleDiameter - this.scoreHeight_;
-
-    render(this.generateTemplate_(), this);
-    this.cursor_ = /** @type {!HTMLElement} */ (this.querySelector("#cursor"));
-    this.score_ = /** @type {!HTMLElement} */ (this.querySelector("#score"));
-
-    if (this.score_ !== null) {
-      this.scoreValueText_ = /** @type {!HTMLElement} */ (this.score_.querySelector(
-        "text#value",
-      ));
-      this.scoreDateText_ = /** @type {!HTMLElement} */ (this.score_.querySelector(
-        "text#date",
-      ));
-    }
-    this.announcer_ = /** @type {!HTMLElement} */ (this.querySelector(
-      ".sr-announcer",
-    ));
-
-    // Animate the line draws by changing the length of the dash-offset.
-    const paths = /** @type {!NodeList<!SVGPathElement>} */ (this.querySelectorAll(
-      ".path:not(.dashed)",
-    ));
-    Array.from(paths).forEach((path) => {
-      const length = path.getTotalLength();
-      path.style.strokeDasharray = length;
-      path.style.strokeDashoffset = length;
+  onResize() {
+    // When the browser window resizes, force a render to redraw the SVG.
+    this.requestUpdate().then(() => {
+      this.redrawPoint();
     });
+  }
 
-    // nb. this relies on side effects of generateTemplate_ which calls
-    // generatePath_ and creates the datapoints array.
-    this.setAttribute("aria-valuenow", this.datapoints.slice(-1)[0].score);
-
-    if (this.fill) {
-      window.requestAnimationFrame(() => {
-        const gradients = this.getElementsByClassName("gradient");
-        for (const gradient of gradients) {
-          gradient.classList.add("fadein");
-        }
-      });
-    }
+  firstUpdated() {
+    this.cursorElement_ = this.renderRoot.querySelector("#cursor");
+    this.scoreElement_ = this.renderRoot.querySelector("#score");
+    this.announcerElement_ = this.querySelector(".sr-announcer");
```

Comment:
maybe we should turn this into an id as well?

Reply:
as discussed in chat—no!